File: backend/internal/model/recipe.go
package model

import "time"

type Recipe struct {
	ID             int       `db:"id"              json:"id"`
	Name           string    `db:"name"            json:"name"`
	EquipmentType  string    `db:"equipment_type"  json:"equipment_type"`
	TargetTemp     float64   `db:"target_temp"     json:"target_temp"`
	TargetPressure float64   `db:"target_pressure" json:"target_pressure"`
	DurationMin    int       `db:"duration_min"    json:"duration_min"`
	CreatedAt      time.Time `db:"created_at"      json:"created_at"`
}

File: frontend/src/api/recipe.ts
// Recipe 相關 API 呼叫封裝
import axios from 'axios'

const BASE = '/api/v1'

export interface Recipe {
  id: number
  name: string
  equipment_type: string
  target_temp: number
  target_pressure: number
  duration_min: number
}

export async function getRecipes(): Promise<Recipe[]> {
  const { data } = await axios.get<{ data: Recipe[] }>(`${BASE}/recipes`)
  return data.data ?? []
}
